Refactoring

fwpushan opened the pull request.

jleach commented on server/router/routes/plan.js:
```diff
@@ -93,15 +93,27 @@ const createAndUpdatePastures = async (plan, body) => {
   // this doens't works since the client doesn't send pastures with id(this can be used later)
   // await Promise.all(pastures.map(async (p) => {
-  //   const { pasture, isCreated } = await Pasture.upsert(p, { returning: true });
+  //   const {
```
Lets cleanup dead code as we go.

Kyubinhan replied:
Oh I left this block because it can be used in the future

fwpushan committed:
Add plan route refactoring

fwpushan committed:
Complete refactoring of plan route
Test with Postman
Changes to be committed:
	modified:   src/libs/db2/model/model.js
	modified:   src/router/controllers/PlanController.js
	new file:   src/router/controllers/PlanInvasivePlantController.js
	new file:   src/router/controllers/PlanManagementConsiderationController.js
	new file:   src/router/controllers/PlanMinisterIssueActionController.js
	modified:   src/router/controllers/PlanMinisterIssueController.js
	modified:   src/router/routes/plan.js

micheal-w-wells reviewed (approved):
looks good but we both will have to have a postman party making sure this behaves as expected before this goes to prod

fwpushan committed:
Fix remove related issue

Commit:
create findWithFilter helper method in user model

Commit:
fix GET /user unit test

Commit:
modify AllUser method in UserContorller

fwpushan committed:
Creating histroy for any status update

Kyubinhan committed:
Merge pull request #46 from bcgov/enhance/get-users-with-params

Enhance/get users with params -> refactoring
